From c89cc908ba7f9f41aba4f61f58103d0fc64fa19f Mon Sep 17 00:00:00 2001
From: Gustavo Zacarias <gustavo@zacarias.com.ar>
Date: Fri, 16 Jun 2017 08:50:03 +0200
Subject: [PATCH] Fix for binutils 2.23.1

libbfd from binutils 2.23.1+ requires PACKAGE* definitions from autoconf.
Patch from https://fedorahosted.org/dropwatch/ticket/5
Upstream status: new.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>
---
 src/lookup.c     | 2 +-
 src/lookup.h     | 3 +++
 src/lookup_bfd.c | 2 +-
 src/lookup_kas.c | 2 +-
 4 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/lookup.c b/src/lookup.c
index ba54991..809981b 100644
--- a/src/lookup.c
+++ b/src/lookup.c
@@ -27,13 +27,13 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <sys/utsname.h>
-#include <bfd.h>
 #include <string.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 
 #include "lookup.h"
+#include <bfd.h>
 
 extern struct lookup_methods bfd_methods;
 extern struct lookup_methods kallsym_methods;
diff --git a/src/lookup.h b/src/lookup.h
index e6568d8..47a7b56 100644
--- a/src/lookup.h
+++ b/src/lookup.h
@@ -28,6 +28,9 @@
 #include <stdlib.h>
 #include <asm/types.h>
 
+// satisfy PR 14072 in bfd.h
+#define PACKAGE 1
+#define PACKAGE_VERSION 1
 
 /*
  * Initalization routine
diff --git a/src/lookup_bfd.c b/src/lookup_bfd.c
index cc7010b..271d426 100644
--- a/src/lookup_bfd.c
+++ b/src/lookup_bfd.c
@@ -25,13 +25,13 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <sys/utsname.h>
-#include <bfd.h>
 #include <string.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 
 #include "lookup.h"
+#include <bfd.h>
 
 
 static int lookup_bfd_init(void)
diff --git a/src/lookup_kas.c b/src/lookup_kas.c
index 16e639f..556cb62 100644
--- a/src/lookup_kas.c
+++ b/src/lookup_kas.c
@@ -25,7 +25,6 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <sys/utsname.h>
-#include <bfd.h>
 #include <string.h>
 #include <unistd.h>
 #include <sys/types.h>
@@ -33,6 +32,7 @@
 #include <sys/queue.h>
 
 #include "lookup.h"
+#include <bfd.h>
 
 struct symbol_entry {
 	char *sym_name;
-- 
2.1.4

